scripts/c8y-publish-release.sh

```diff
@@ -48,6 +50,13 @@ publish_version() {
     fi
 }
 
+# Set from draft to pre-release
+if ! [[ -z $(gh release list | grep $TAG | grep Draft) ]] && [[ $PRERELEASE = "true" ]]; then 
+    echo "Update Release from Draft to Pre-Release"
+    gh release edit --draft=false --prerelease $TAG 
+    sleep 3
+fi
+
```

Comment:
It would be better to use the `gh` command rather than relying on grep (as this will run into pagination issues if the `gh release list` might not see the tag that you're trying to publish).

```suggestion
if [ "$PRERELEASE" = "true" ]; then
    IS_DRAFT=$(gh release view "$TAG" --json isDraft  --template "{{.isDraft}}")
    if [ "$IS_DRAFT" = "true" ]; then
        gh release edit --draft=false --prerelease "$TAG"
    fi
    sleep 3
fi
```

Reply:
Still not great to use a magic `sleep 3`...It would probably be better to write a small function using the "gh release view" to check what the expected value should be, and then continue.